controllers/v1/entities.js

```diff
@@ -19,26 +20,695 @@
+    *                  "visible": true,
+    *                  "icon": "view"
+    *              }
+    *          ]
+    *      }
+    *  ],
+    *  "data": [
+    *      {
+    *          "externalId": "Dl",
+    *          "name": "New Delhi",
+    *          "_id": "5d6609ef81a57a6173a79e78",
+    *         "childHierarchyPath": [
+    *              "district",
+    *              "zone",
+    *              "school"
+    *          ],
+    *          "createdAt": "28th Aug 2019"
+    *      },
+    *  ]
+    * }
+    * }
+    * 
+  **/
+
+  /**
+    * List of entities based on its type.
+    * @method
+    * @name listByEntityType
+    * @param  {Request} req request body.
+    * @returns {JSON} Returns list of entities
+   */
+
+  listByEntityType(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let requestedData = {
+          entityType: req.params._id,
+          pageSize: req.pageSize,
+          pageNo: req.pageNo,
+          searchText : req.searchText
+        }
+
+        let entityDocuments = await entitiesHelper.listByEntityType(
+          requestedData
+        );
+        return resolve(entityDocuments);
+
+      } catch (error) {
+
+        return reject({
+          status: error.status || httpStatusCode.internal_server_error.status,
+          message: error.message || httpStatusCode.internal_server_error.message,
+          errorObject: error
+        })
+
+      }
+    })
+  }
+
+  /**
+   * @api {post} /admin-service/api/v1/entities/subEntityList/:entityId?type=:type&search=:search&page=:page&limit=:limit
+   * Get sub entity list for the given entity. 
+   * @apiVersion 1.0.0
+   * @apiGroup Entities
+   * @apiHeader {String} X-authenticated-user-token Authenticity token
+   * @apiSampleRequest /admin-service/api/v1/entities/subEntityList/5db173598a8e070bedca6ba1?type=school&search=r&page=1&limit=1
+   * @apiUse successBody
+   * @apiUse errorBody
+   * @apiParamExample {json} Response:
+   * {
+   * "message": "List of entities fetched successfully",
+   * "status": 200,
+   * "result": {
+   *  "data": [
+          {
+              "_id": "5db1dd3e8a8e070bedca6bef",
+              "entityType": "school",
+              "name": "Sachdeva Convent School, Street No.-5 Sangam Vihar (Wazirabad - Jagatpur Road), Delhi",
+              "externalId": "1207229",
+              "addressLine1": "Street No.-5 Sangam Vihar (Wazirabad - Jagatpur Road)",
+              "label": "Sachdeva Convent School, Street No.-5 Sangam Vihar (Wazirabad - Jagatpur Road), Delhi",
+              "value": "5db1dd3e8a8e070bedca6bef"
+          }
+      ],
+      "count": 6005
+  }
+}
+}
+
+  /**
+    * Get the immediate entities .
+    * @method
+    * @name subEntityList
+    * @param  {Request} req request body.
+    * @returns {JSON} Returns list of immediate entities
+   */
+
+  subEntityList(req) {
+    return new Promise(async (resolve, reject) => {
+
+      if (!(req.params._id)) {
```

Comment:
@rakeshSgr - Please check the same condition along with a mongo id in validator file